packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -265,10 +266,21 @@ const ForwardedInnerBlocks = forwardRef( ( props, ref ) => {
 	};
 
 	// Detects if the InnerBlocks should be controlled by an incoming value.
-	if ( props.value && props.onChange ) {
-		return <ControlledInnerBlocks { ...allProps } />;
-	}
-	return <UncontrolledInnerBlocks { ...allProps } />;
+	return (
+		<Disabled.Consumer>
```

Comment:
Why is this necessary? It seems for me that this is just hiding another error that may happen even if the  component is not disabled.

Reply:
You’re right that the underlying bug also needs fixed, but I feel like it is still worth adding this protection. At the top level block editor provider, we can simply not attach an onChange handler, but in a nested template part, that isn’t an option. If any other bugs in the future ever do arise which end up calling onChange or onInput while the component is supposed to be disabled, they can cause the entity which powers the block to be entirely wiped of its content.

I feel like it’s a good idea to guard against that possibility, since it would never be a user expectation for an entity to get modified at all when loading a block preview, for example.

Reply:
Shouldn't that be guarded with a test instead?

Reply:
I think "Disabled" doesn't convey the meaning that we don't want "onChange" to be called. It's just a UI component. If we were to introduce such thing I'd opt for a "preview mode" where we could disable a lot of things  including the onChange calls, but that seems like a big  undertaking for now.

Reply:
I see. I think that’s fair enough. I do disagree because I think disabled does convey the meaning. Plus, I imagine there are situations outside of block preview where I would want to make sure changes are unidirectional from entity to blocks and not the other way around... e.g. whenever a controlled inner block is nested within the Disabled component. 

Per tests, I’m not convinced we can guard against every case. If another similar bug does arise, we can guarantee that it will unexpectedly modify an entity, perhaps even overwriting it, but we can’t guarantee our test case will cover that situation.

Reply:
All the test needs to do is mount it and check that there are no edits.